proposals/pr001-ecma402.md

```diff
@@ -0,0 +1,208 @@
+  matures, implementation can be switched.
+
+* It is as good a time as any to design this API for rust.
+
+  It should not matter that ICU4X in particular has not yet defined its
+  ergonomic APIs. The ECMA-402  surface should not depend on what is convenient
+  any specific implementing library, but rather should be ergonomic to use.
+
+## Goal
+
+_Validate the concept of a common ECMA 402 API, or fail quickly._
+
+This doc is submitted for comments as a sanity check on the approach.  Question
+to you, the potential commenter: **do you think this exercise is worth doing?**
+
+
+## Landscape
+
+Rust's support for internationalized text handling is "[still very immature and
+unstable, with most libraries being only weeks old and not used in production
+yet](https://www.arewewebyet.org/topics/i18n/)".  The [ECMA-402 support feature
+request for Rust](https://github.com/rust-lang/rfcs/issues/858) has been open
+for 5 years, with currently no resolution.
+
+I suspect two factors contribute:
+
+*   **The enormous barrier to entry.**
+
+	Writing internationalized text support requires specific knowledge of
+	internationalization, as well as the many corner cases of particular
+	languages.  Such knowledge typically exceeds the knowledge of any one
+	person, making an organized team effort practically a requirement to make
+	progress.
+
+*   **Relatively niche interest.**
+
+	Anecdotally it seems to me that, when it comes to internationalization,
+	people are in general ready to scale down their aspirations if there is no
+	canned solution to their problem.  But if you _really _need it, then there
+	is no fallback solution for you.
+
+So, the prophecy is self-fulfilling: we'd be glad to use a library if it
+existed; but we are unwilling to contribute because it's expensive to make.
+The status quo continues.
+
+I'll present an assumption: **if robust text internationalization support were
+available for rust today, then people who need it would use it in their rust
+programs**.  I tested this assumption with
+[rust\_icu](https://github.com/google/rust_icu), a rust wrapper for the
+[Unicode's ICU library](http://site.icu-project.org/home).  While the jury is
+still out as to where this goes, given that the library is still [relatively
+unknown](https://www.google.com/search?q=rust+unicode+support), and used mostly
+only in Fuchsia OS, it **makes it cheap** to contribute internationalized text
+support thanks to ICU providing the heavy lifting.  To illustrate: it took me a
+morning and half an afternoon to provide [collation (locale-sensitive
+sorting)](https://crates.io/crates/rust_icu_ucol) support for rust building on
+top of `rust_icu`.  There were no rust collation libraries prior to that
+afternoon.
+
+In parallel, ICU4X is a project that will provide next-generation support for
+internationalized text, based on a Rust implementation of Unicode standards.
+This project is still some time away from first usable functionality, is my
```

Comment:
I would make whole proposal less personal and more fact based:

This project is still in early phase of development and not ready for general use. (no need for my estimate, first usable functionality etc).